public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -766,6 +766,8 @@ export function buildExpr(
 }
 
 export function buildLinkExpr(expr: string) {
+  // don't want top 5 or by span name in links
+  expr = expr.replace('topk(5, ', '').replace(' by (span_name))', '');
```

Comment:
unlikely but what happens if someone applies a filter with one of these substrings. i.e. at the top they have selected something silly like `nameapace="topk(5, "`

Reply:
We wouldn't reach this part of the code as the query wouldn't return any frames. Ideally we should not let users input anything into the filter inputs other that what is already in the dropdown?